tfjs-backend-webgpu/src/shader_preprocessor_wgsl.ts

```diff
@@ -245,48 +245,48 @@ const SHADER_PREFIX = `
   }
 
   // Checks whether coordinates lie within the bounds of the shape.
-  fn coordsInBounds4D(coord : vec4<u32>, shape : vec4<u32>) -> bool {
+  fn coordsInBounds4D(coord : vec4<i32>, shape : vec4<i32>) -> bool {
     return all(coord < shape);
```

Comment:
Should your previous changes be reverted #5559? Use `return all(coord >= vec4<u32>(0u)) && all(coord < shape);`. And similar for other places.